### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by shiimaxx**
This PR improves CI to simplify.

I added Makefile and defined some tasks like imports(goimports), lint(golint), test(go test), and build_cross(cross-build by goxz).

These make tasks replaces the build step in wercker.yml and can run in the local machine as well by make.

## Changes in wercker.yml

Changes

- "goimports", "golint", "go test"
    - change to execute make task

Deletes

- "go vet"
    - `go test` runs `go vet` automaticaly from Go1.10
        - https://golang.org/doc/go1.10#test
- "install test plugin"
    - it's unused

## Artifacts of CI

I changed the way of cross-build to using goxz in this PR, so the artifacts are changes following.

### Before

```
./dist/MD5SUM
./dist/linux_amd64.tar.gz
./dist/windows_amd64.zip
```

### After

```
./dist/MD5SUM
./dist/happo-agent_<VERSION>_darwin_amd64.zip
./dist/happo-agent_<VERSION>_linux_amd64.tar.gz
./dist/happo-agent_<VERSION>_windows_amd64.zip
````

### CODE REVIEW

File: wercker.yml [hunk 1]
```diff
@@ -51,13 +34,13 @@ build:
             ${BINARY:?} remove -h
             ${BINARY:?} append_metric -h
             ${BINARY:?} help
-    - script:
+    - script:
```
Comment (netmarkjp) [inline wercker.yml]: Please remove trailing whitespace.

**Comment by netmarkjp** [comment]
@shiimaxx Thanks a lot :relaxed: 
I felt that this PR is a very good improvement.

I have one question. Is Zip the convenient archive format for darwin?

File: wercker.yml (hunk 1 above)
Comment (shiimaxx) [reply wercker.yml]: I removed it, and other spaces too.

**Comment by shiimaxx** [comment]
@netmarkjp 
Thanks for the review.

> Is Zip the convenient archive format for darwin?

zip is default installed in Darwin(macOS), so I think can handle without the inconvenience

**Pull request merged**